ghproxy/ghmetrics/ghmetrics.go

```diff
@@ -0,0 +1,61 @@
+package ghmetrics
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+// ghTokenUsageGaugeVec provides the 'github_token_usage' gauge that
+// enables keeping track of GitHub calls and quotas.
+var ghTokenUsageGaugeVec = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "github_token_usage",
+		Help: "How many GitHub token requets have been used up.",
+	},
+	[]string{"remaining", "until_reset"},
+)
+
+// ghRequestsGauge provides the 'github_requests' gauge that keeps track
+// of the number of GitHub requests by API path.
+var ghRequestsGauge = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "github_requests",
+		Help: "GitHub requests by API path.",
+	},
+	[]string{"path", "status", "duration"},
+)
+
+func init() {
+	prometheus.MustRegister(ghTokenUsageGaugeVec)
+	prometheus.MustRegister(ghRequestsGauge)
+}
+
+// CollectGithubTokenMetrics publishes the rate limits of the github api to
+// `github_token_usage` on prometheus.
+func CollectGithubTokenMetrics(headers http.Header, now time.Time) {
+	remaining := headers.Get("X-RateLimit-Remaining")
+	timeUntilReset := timeUntilFromUnix(headers.Get("X-RateLimit-Reset"), now)
+
+	ghTokenUsageGaugeVec.With(prometheus.Labels{"remaining": remaining, "until_reset": timeUntilReset.String()}).Inc()
```

Comment:
Instead of a gauge incremented to count the number of times we update this, shouldn't this set the gauge value to the remaining tokens?